tests/tests_data/streaming/test_dataset.py

```diff
@@ -0,0 +1,201 @@
+# Copyright The Lightning AI team.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import pytest
+import torch
+from lightning import seed_everything
+from lightning.data.datasets.env import _DistributedEnv
+from lightning.data.streaming import Cache
+from lightning.data.streaming import cache as cache_module
+from lightning.data.streaming.dataloader import StreamingDataLoader
+from lightning.data.streaming.dataset import StreamingDataset
+from lightning.data.streaming.item_loader import TokensLoader
+from lightning.data.streaming.shuffle import AcrossChunkShuffle, MinShuffle, NoShuffle
+from lightning.pytorch.demos.boring_classes import RandomDataset
+from torch.utils.data import DataLoader
+
+
+def test_streaming_dataset(tmpdir, monkeypatch):
+    seed_everything(42)
+
+    os.makedirs(os.path.join(tmpdir, "remote_dir"), exist_ok=True)
+    monkeypatch.setattr(cache_module, "_try_create_cache_dir", lambda name: tmpdir)
+
+    with pytest.raises(ValueError, match="The provided dataset `choco` isn't filled up."):
+        dataset = StreamingDataset(name="choco", cache_dir=tmpdir)
+
+    dataset = RandomDataset(128, 64)
+    dataloader = StreamingDataLoader(dataset, cache_dir=tmpdir, chunk_bytes=2 << 12)
+    for batch in dataloader:
+        assert isinstance(batch, torch.Tensor)
+
+    dataset = StreamingDataset(name="choco", cache_dir=tmpdir, item_loader=TokensLoader(block_size=10))
+
+    assert len(dataset) == 816
+    dataset_iter = iter(dataset)
+    assert len(dataset_iter) == 816
+
+    dataloader = DataLoader(dataset, num_workers=2, batch_size=2)
+    assert len(dataloader) == 408
+
+
+def test_streaming_dataset_distributed_min_shuffle(tmpdir):
```

Comment:
this test doesn't test the effect of shuffling, only that the end result is correctly partitioned

Reply:
Added some extra bits.